docs/kafka/topic-configuration-kafka/README.adoc

```diff
@@ -352,6 +354,28 @@ h|Kafka property name
 |===
 --
 
+Message down-conversion::
```

Comment:
@showuon Sure thing. But just to clarify again, my specific confusion is that we are with this update both removing the `message.format.version` and adding `message.downconversion.enable` set to `true` by default, which therefore hinges on the message format property being up to date or not, which we're removing and no allowing users to update. So it's just kind of odd and a bit of circular logic from what I'm understanding. Thanks.

Reply:
I'm not sure I got your idea. But here's my thought:

> My understanding is that as an alternative to using this down-conversion property, users can just ensure that their clients get upgraded to the correct message.format.version to avoid down-conversion

Yes, that's true, if the users upgrade their producer first to allow newer version of `message.format.version`. But that's not under our control. Besides, there might be some other cases we don't know that needs down-conversion. 
From my understanding, the disadvantage of allowing down-conversion is it will consume CPU resources. But on the other hand, if provide flexibility to users when they need to consume old message versions.

My thought is:
1. Maybe you're right, since we don't allow users to edit `message.format.version`, there will not have `down-conversion` happen. If that's true, then this down-conversion property won't hurt our server
2. If there are cases that user still needs to do down-conversion, we can find it if the CPU load is high, and then change the limit. But before that, I think providing the flexibility to users is a better option.

Does that make sense?